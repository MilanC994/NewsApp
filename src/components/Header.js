import React from 'react'
import { Navbar, Form, Button, FormControl } from 'react-bootstrap'
import useHeader from './hooks/useHeader'

const Header = () => {
  const { search, selectList } = useHeader()

    return (
        <Navbar fixed="top" className="mb1" bg="primary" variant="dark" expand="md">
        <Navbar.Brand href="#">News Application</Navbar.Brand>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Form inline className="ml-auto" onSubmit={(e)=>{e.preventDefault()}}>
            <FormControl type="text" placeholder="Search" className="mr-sm-2" { ...search }/>
            <Form.Control
              as="select"
              className="mr-sm-2" 
              id="inlineFormCustomSelect"
              custom
              {...selectList.item}
            >
              {selectList.options.map(item => <option key={item.value} {...item}>{item.name}</option>)}
            </Form.Control> 
          </Form>
            {/* <Nav style={{ border:"2px solid red" }}>
              <Nav.Link href="#home">Home</Nav.Link>
              <Nav.Link href="#link">Link</Nav.Link>
              <NavDropdown title="Dropdown" id="basic-nav-dropdown">
                <NavDropdown.Item href="#action/3.1">Action</NavDropdown.Item>
                <NavDropdown.Item href="#action/3.2">Another action</NavDropdown.Item>
                <NavDropdown.Item href="#action/3.3">Something</NavDropdown.Item>
                <NavDropdown.Divider />
                <NavDropdown.Item href="#action/3.4">Separated link</NavDropdown.Item>
              </NavDropdown>
            </Nav> */}
            
        </Navbar.Collapse>
      </Navbar>
        )
}

export default Header
