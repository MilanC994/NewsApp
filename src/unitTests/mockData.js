import { createStore, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'
import reducer from '../redux/reducer'

export const createTestStore = initialState => {
  const createStoreWithMiddleware = applyMiddleware(thunk)(createStore)
  return createStoreWithMiddleware(reducer, initialState)
}
export const mockArticles = [
  {
    author: 'Jim Halpert',
    content: 'test content 1',
    description: 'test description 1',
    publishedAt: '2021-01-02T13:25:54Z',
    source: { id: null, name: 'test src 1' },
    title: 'test title 1',
    url: 'test site url 1',
    urlToImage: 'test image url 1',
  },
  {
    author: 'Pam Beasley',
    content: 'test content 2',
    description: 'test description 2',
    publishedAt: '2021-01-02T13:25:54Z',
    source: { id: null, name: 'test src 2' },
    title: 'test title 2',
    url: 'test site url 2',
    urlToImage: 'test image url 2',
  },
  {
    author: 'Stanley Hudson',
    content: 'test content 3',
    description: 'test description 3',
    publishedAt: '2021-01-02T13:25:54Z',
    source: { id: null, name: 'test src 3' },
    title: 'test title 3',
    url: 'test site url 3',
    urlToImage: 'test image url 3',
  },
  {
    author: 'Dwight Schrute',
    content: 'test content 4',
    description: 'test description 4',
    publishedAt: '2021-01-02T13:25:54Z',
    source: { id: null, name: 'test src 4' },
    title: 'test title 4',
    url: 'test site url 4',
    urlToImage: 'test image url 4',
  },
  {
    author: 'Creed Bratton',
    content: 'test content 5',
    description: 'test description 5',
    publishedAt: '2021-01-02T13:25:54Z',
    source: { id: null, name: 'test src 5' },
    title: 'test title 5',
    url: 'testsiteurl5',
    urlToImage: 'testimageurl5',
  },
]

export const mockResponse = {
  status: 200,
  statusText: '',
  data: {
    status: 'ok',
    totalResults: 5,
    articles: mockArticles,
  },
}

export const mockState = {
  articles: [],
  searchTerm: '',
  sortBy: 'publishedAt',
  country: 'us',
  page: 1,
  totalResults: 35,
}
