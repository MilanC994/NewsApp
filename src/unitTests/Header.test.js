import { createTestStore } from './mockData'
import React from 'react'
import {  mount } from 'enzyme'
import { mountToJson } from 'enzyme-to-json'
import Header from '../components/Header'
import { Provider } from 'react-redux'

// const setUp = (initialState, expectedValue) => {
//     const store = createTestStore(initialState) 

//     const wrapper = mount(
//         <Provider store={store}>
//         <Header />
//         </Provider>
//     ).childAt(0)

//     const element = wrapper.find('.countries-sort-select-list').at(1)
//     expect(element.props().value).toEqual(expectedValue)
// }
const setUp = (initialState) => {
    const store = createTestStore(initialState) 

    const wrapper = mount(
                        <Provider store={store}>
                            <Header />
                        </Provider>
                    ).childAt(0)
    return wrapper
}

describe('Header test', () => {
    //TODO - figure out how to pass parameters to beforeEach
    test('Renders correctly', () => {
        const initialState = {
            articles: [],
            searchTerm: '',
            sortBy: 'publishedAt',
            country:'us',
            page: 1,
            totalResults: null
        }
        const wrapper = setUp(initialState)
        expect(mountToJson(wrapper)).toMatchSnapshot()
            })
    
    test('Renders countries in select list when search bar is empty', () => {
        const initialState = {
            articles: [],
            searchTerm: '',
            sortBy: 'publishedAt',
            country:'us',
            page: 1,
            totalResults: null
        }
        const wrapper = setUp(initialState)
        const element = wrapper.find('.countries-sort-select-list').at(1)
        expect(element.props().value).toEqual('us')
       
    })
    test('Renders sort by in select list when search bar is not empty', () => {
    
        const initialState = {
            articles: [],
            searchTerm: 'the office',
            sortBy: 'publishedAt',
            country:'us',
            page: 1,
            totalResults: null
        }
        const wrapper = setUp(initialState)
        const element = wrapper.find('.countries-sort-select-list').at(1)
        expect(element.props().value).toEqual('publishedAt')
    })
})