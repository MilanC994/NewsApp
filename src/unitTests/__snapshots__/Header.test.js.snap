// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test Renders correctly 1`] = `
<Header>
  <Navbar
    bg="primary"
    className="mb1"
    collapseOnSelect={false}
    expand="md"
    fixed="top"
    variant="dark"
  >
    <nav
      className="mb1 navbar navbar-expand-md navbar-dark bg-primary fixed-top"
    >
      <NavbarBrand
        href="#"
      >
        <a
          className="navbar-brand"
          href="#"
        >
          News Application
        </a>
      </NavbarBrand>
      <NavbarToggle
        aria-controls="basic-navbar-nav"
        label="Toggle navigation"
      >
        <button
          aria-controls="basic-navbar-nav"
          aria-label="Toggle navigation"
          className="navbar-toggler collapsed"
          onClick={[Function]}
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
      </NavbarToggle>
      <NavbarCollapse
        id="basic-navbar-nav"
      >
        <ForwardRef
          appear={false}
          getDimensionValue={[Function]}
          id="basic-navbar-nav"
          in={false}
          mountOnEnter={false}
          timeout={300}
          unmountOnExit={false}
        >
          <Transition
            addEndListener={[Function]}
            appear={false}
            aria-expanded={null}
            enter={true}
            exit={true}
            id="basic-navbar-nav"
            in={false}
            mountOnEnter={false}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            timeout={300}
            unmountOnExit={false}
          >
            <div
              aria-expanded={null}
              className="navbar-collapse collapse"
              id="basic-navbar-nav"
            >
              <Form
                className="ml-auto"
                inline={true}
                onSubmit={[Function]}
              >
                <form
                  className="ml-auto form-inline"
                  onSubmit={[Function]}
                >
                  <FormControl
                    className="mr-sm-2"
                    onChange={[Function]}
                    placeholder="Search"
                    type="text"
                    value=""
                  >
                    <input
                      className="mr-sm-2 form-control"
                      onChange={[Function]}
                      placeholder="Search"
                      type="text"
                      value=""
                    />
                  </FormControl>
                  <FormControl
                    as="select"
                    className="mr-sm-2 countries-sort-select-list"
                    custom={true}
                    id="inlineFormCustomSelect"
                    onChange={[Function]}
                    value="us"
                  >
                    <select
                      className="mr-sm-2 countries-sort-select-list custom-select"
                      id="inlineFormCustomSelect"
                      onChange={[Function]}
                      value="us"
                    >
                      <option
                        key="us"
                        name="United States"
                        value="us"
                      >
                        United States
                      </option>
                      <option
                        key="ca"
                        name="Canada"
                        value="ca"
                      >
                        Canada
                      </option>
                      <option
                        key="gb"
                        name="Great Britain"
                        value="gb"
                      >
                        Great Britain
                      </option>
                      <option
                        key="nz"
                        name="New Zealand"
                        value="nz"
                      >
                        New Zealand
                      </option>
                      <option
                        key="au"
                        name="Australia"
                        value="au"
                      >
                        Australia
                      </option>
                      <option
                        key="de"
                        name="Germany"
                        value="de"
                      >
                        Germany
                      </option>
                    </select>
                  </FormControl>
                </form>
              </Form>
            </div>
          </Transition>
        </ForwardRef>
      </NavbarCollapse>
    </nav>
  </Navbar>
</Header>
`;
